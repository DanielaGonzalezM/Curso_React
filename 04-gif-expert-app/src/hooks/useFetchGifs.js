import { useEffect, useState } from 'react';
import { getGift } from '../helpers/getGifs'


export const useFetchGift = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });
    useEffect(() => {
        getGift(category
        ).then(img => {

                setState({
                    data: img,
                    loading: false
                })


        })

    }, [category]);

    return state;
}