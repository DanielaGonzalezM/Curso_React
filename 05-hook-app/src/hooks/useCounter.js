import {useState} from 'react';


export const useCounter = (initial = 10) => {

    const [state, setState] = useState(initial);

    const increment = (factor=1) => {
        setState(state + factor);
    };

    const decrement = (factor=factor) => {
        setState(state - 1);
    };

    const reset = () => {
        setState(initial);
    };


    return {state, increment, decrement, reset};

}
