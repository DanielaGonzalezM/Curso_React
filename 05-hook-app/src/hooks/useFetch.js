import { useEffect, useRef, useState } from "react";

export const useFetch = (url) => {
  const isMounted = useRef(true);
  const [state, setState] = useState({
    data: null,
    loading: true,
    error: null,
  });
  useEffect(() => {
    return () => {
      isMounted.current = false;
    };
  }, []);
  useEffect(() => {
    setState({ loading: true, error: null, data: null });
    fetch(url)
      .then((resp) => resp.json())
      .then((data) => {
        if (isMounted.current) {
          setState({ loading: false, error: null, data: data });
        }
      })
      .catch(() => {
        setState({
          loading: false,
          error: "No se pudo cargar la info",
          data: null,
        });
      });
  }, [url]);
  return state;
};
