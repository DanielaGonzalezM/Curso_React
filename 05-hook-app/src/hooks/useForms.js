import {useState} from "react"


export const useForms = (initialState = {}) => {

    const [values, setvalues] = useState(initialState);

    const handleInputChange = ({target}) => {

        setvalues({
            ...values,
            [target.name]: target.value
        });

    }

    return [values, handleInputChange];
}
