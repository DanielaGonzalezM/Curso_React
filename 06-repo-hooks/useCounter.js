import {useState} from 'react';


export const useCounter = (initial = 10) => {

    const [cont, setState] = useState(initial);

    const increment = () => {
        setState(cont + 1);
    };

    const decrement = () => {
        setState(cont - 1);
    };

    const reset = () => {
        setState(initial);
    };


    return { cont, increment, decrement, reset};

}
