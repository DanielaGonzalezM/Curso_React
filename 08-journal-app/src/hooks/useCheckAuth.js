import { onAuthStateChanged } from "firebase/auth";
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { FirebaseAuth } from "../firebase/config";
import { login, logout } from "../store/auth";
import { startLoadingNotes } from "../store/journal/thunks";

export const useCheckAuth = () => {
  const { status } = useSelector((state) => state.auth);
  const dispatch = useDispatch();
  useEffect(() => {
    onAuthStateChanged(FirebaseAuth, async (user) => {
      if (!user) return dispatch(logout());
      const { uid, email, displayName, photoURL } = user;
      dispatch(login({ uid, email, displayName, photoURL }));
      dispatch(startLoadingNotes());
    });
  }, []);
  return { status };
};
