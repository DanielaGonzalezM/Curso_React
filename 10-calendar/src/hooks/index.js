export * from "./useUiStore";
export * from "./useCalendarStore";
