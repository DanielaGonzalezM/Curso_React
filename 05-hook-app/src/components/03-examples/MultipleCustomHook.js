import React from "react";
import { useCounter } from "../../hooks/useCounter";
import { useFetch } from "../../hooks/useFetch";
import "../02-useEffect/effects.css";
const MultipleCustomHook = () => {
    
   const {cont,increment}=useCounter(1);

  const { loading, data } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${cont}`
  );
  const { author, quote } = !!data && data[0];
  return (
    <div>
      <h1>BreakingBad Quites</h1>
      <hr />
      {loading ? (
        <div className="alert alert-info text-center">loading...</div>
      ) : (
        <blockquote className="blockquote text-end">
          <p className="mb-2"> {quote}</p>
          <footer className="blockquote-footer"> {author}</footer>
        </blockquote>
      )}
      <button className="btn btn-primary" onClick={increment}>
          Siguiente quote
      </button>
    </div>
  );
};

export default MultipleCustomHook;
