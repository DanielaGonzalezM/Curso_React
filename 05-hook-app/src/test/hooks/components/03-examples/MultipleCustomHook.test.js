import { shallow } from "enzyme";
import MultipleCustomHook from "../../../../components/03-examples/MultipleCustomHook";
import { useCounter } from "../../../../hooks/useCounter";
import { useFetch } from "../../../../hooks/useFetch";
jest.mock("../../../../hooks/useFetch");
jest.mock("../../../../hooks/useCounter");
describe("Pruebas en MultipleCustomHook", () => {
  
  test("debe mostrarse correctamente", () => {
    useCounter.mockReturnValue({
      cont: 10,
      increment: () => {},
    });
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHook />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar informaciÃ³n", () => {
    useCounter.mockReturnValue({
      cont: 10,
      increment: () => {},
    });
    useFetch.mockReturnValue({
      data: [
        {
          author: "El cuervo",
          quote: "Todo hombre tiene un diablo y no descansa hasta encontrarlo",
        },
      ],
      loading: false,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHook />);
    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-2").text().trim()).toBe(
      "Todo hombre tiene un diablo y no descansa hasta encontrarlo"
    );
    expect(wrapper.find("footer").text().trim()).toBe("El cuervo");
  });
});
