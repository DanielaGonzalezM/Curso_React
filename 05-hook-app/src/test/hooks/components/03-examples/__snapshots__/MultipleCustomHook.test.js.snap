// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultipleCustomHook debe mostrarse correctamente 1`] = `
<div>
  <h1>
    BreakingBad Quites
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
